src/R/Support/Impl/Help/IntelliSenseRSession.cs

```diff
@@ -79,11 +96,54 @@ public sealed class IntelliSenseRSession : IIntellisenseRSession {
 
                 if (!Session.IsHostRunning) {
                     int timeout = _coreShell.IsUnitTestEnvironment ? 10000 : 3000;
-                    await Session.EnsureHostStartedAsync(new RHostStartupInfo (RToolsSettings.Current.CranMirror, codePage: RToolsSettings.Current.RCodePage), null, timeout);
+                    await Session.EnsureHostStartedAsync(new RHostStartupInfo(RToolsSettings.Current.CranMirror, codePage: RToolsSettings.Current.RCodePage), null, timeout);
                 }
             } finally {
                 token.Set();
             }
         }
+
+        /// <summary>
+        /// Retrieves names of packages loaded into the interactive session.
+        /// </summary>
+        public IEnumerable<string> LoadedPackageNames {
+            get {
+                if (_loadedPackages == null) {
+                    if (_workflow.RSession != null) {
+                        _workflow.RSession.Mutated += OnInteractiveSessionMutated;
+                        UpdateListOfLoadedPackagesAsync().Wait(2000);
+                    }
+                }
+                return _loadedPackages ?? Enumerable.Empty<string>();
+            }
+        }
+
+        private void OnInteractiveSessionMutated(object sender, EventArgs e)
+             => UpdateListOfLoadedPackagesAsync().DoNotWait();
+
+        private async Task UpdateListOfLoadedPackagesAsync() {
+            try {
+                await StartSessionAsync();
+                var session = GetLoadedPackagesInspectionSession();
+                if (session != null) {
+                    var response = await session.EvaluateAsync<string>("paste0(.packages(), collapse = ' ')", REvaluationKind.Normal);
```

Comment:
Remember that you can always quote identifiers with backticks - ``library(`foo bar`)`` etc. I don't think that's the standard naming convention, but users can ignore naming conventions, so it's still potential input that can be handled.

Anyway, I think it's just simpler to treat it as an array all around, since we have the code to support marshaling it in JSON serializer - you can just do `EvaluateAsync<string[]>()` and get the values back.